vignettes/pr_review_guidance.Rmd

```diff
@@ -0,0 +1,97 @@
+---
+title: "Pull Request Review Guidance"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{pr_review_guidance}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Introduction
+
+This document is intended to be a guidance for reviewers of pull requests (PRs) in the {admiral} package. PR authors will benefit from shorter review times by closely following the guidance provided here.
+
+
+# Review Criteria
+
+For a pull request to be merged it needs to pass the automated `R CMD check` and `lintr` workflows on GitHub at a minimum. These checks can also be run locally using the `devtools::check()` and `lintr::lint_package()` commands, respectively. In addition the PR reviewer should make sure that
+
+- the [Programming Strategy](programming_strategy.html) is followed
+
+- the function is ADaM IG compliant
+
+- the function does what is intended for (as described in the header and corresponding issue)
+
+- the function header properly explains the function
+
+- the implemented derivation is in the scope of admiral, e.g. does not expect company specific input or hard-code company-specific rules
+
+- meaningful error or warning messages are issued if the input is invalid
+
+- documentation is created/updated by running `devtools::document()`
+
+- functions which are supposed to be exported are listed in the `NAMESPACE` file; this requires an `@export` tag in the function header
+
+- examples print relevant source variables and newly created variables and/or records in their output
+
+- the `NEWS.md` file is updated with an entry that explains the new features or changes
+
+- the author of a function is listed in the `DESCRIPTION` file
+
+- all files affected by the implemented changes, e.g. vignettes and templates, are updated
+
```

Comment:
I would add that the test coverage must be at least 90%.

Reply:
In the meeting we decided for "aim for 90% for derive functions". However I think a guidance regarding the other functions and the whole package would be helpful.

For the new functions I am implementing I achieve 100% coverage with reasonable effort.

Reply:
Updated.